src/ISFirmwareUpdater.cpp

```diff
@@ -105,7 +105,20 @@ bool ISFirmwareUpdater::fwUpdate_handleDone(const fwUpdate::payload_t &msg) {
     return true;
 }
 
-fwUpdate::msg_types_e ISFirmwareUpdater::fwUpdate_step() {
+bool ISFirmwareUpdater::fwUpdate_isDone()
+{
+    if (hasPendingCommands())
```

Comment:
While this is easier to read, it should probably just be a series of if ( () || () || () || () ) return false;